index.js

```diff
@@ -11,11 +12,41 @@ server.get('/payments', (req, res) => {
     res.status(200).send(payments);
 });
 
+server.get('/payments/:id', (req, res) => {
+    const payment = payments.find( payment => payment.id === req.params.id);
+    if(payment){
+        res.status(200).send(payment);
+    }
+    else{
+        res.status(404).send({message: 'Wrong id'});
+    }
+});
+
+server.post('/payments', (req, res) => {
+    const payment = {...req.body};
+    Object.assign(payment, {
+        id: uuidv4(),
+        status: 'Pending'
+    });
+    payments.push(payment);
+    res.status(200).send(payment);
+});
+
+server.put('/payments/:id', (req, res) => {
+    const payment = payments.find( payment => payment.id === req.params.id);
+    payment.description = req.body.description;
+    res.status(200).send(payment);
+});
 
+server.delete('/payments/:id', (req, res) => {
+    const payment = payments.find( payment => payment.id === req.params.id);
+    const paymentIdx = payments.indexOf(payment);
+    payments.splice(paymentIdx, 1);
+    res.status(200).send(payments);
```

Comment:
hi, 
when we delete a payment, sending OK status is enough:
```js
res.status(200).send();
```
or if we want to give more info, then it might be reasonable to send the deleted payment only. `Arrays.splice()` method returns and array of deleted items. so you could probably send that instead. but, keep in mind: we deleted by id, which means: we always only delete a single payment, not an array of payments.